python/pyspark/rdd.py

```diff
@@ -2923,11 +3297,38 @@ def getResourceProfile(self):
         else:
             return None
 
+    @overload
+    def toDF(
+        self: "RDD[RowLike]",
+        schema: Optional[Union[List[str], Tuple[str, ...]]] = None,
+        sampleRatio: Optional[float] = None,
+    ) -> "DataFrame":
+        ...
+
+    @overload
+    def toDF(
+        self: "RDD[RowLike]", schema: Optional[Union["StructType", str]] = None
+    ) -> "DataFrame":
+        ...
+
+    @overload
+    def toDF(
+        self: "RDD[AtomicValue]",
+        schema: Union["AtomicType", str],
+    ) -> "DataFrame":
+        ...
+
+    def toDF(
+        self: "RDD[Any]", schema: Optional[Any] = None, sampleRatio: Optional[float] = None
+    ) -> "DataFrame":
+        raise RuntimeError("""RDD.toDF was called before SparkSession was initialized.""")
```

Comment:
I am not very happy about this, but as far as I can tell it is the only way to type check `toDF`.

Reply:
Just noticed that [this](https://github.com/python/mypy/pull/11944) was merged. So I guess we can  wait for upgrade to mypy to dev and drop the implementation.